src/v1/types/requests.rs

```diff
@@ -47,17 +47,22 @@ pub struct ListTasksParams {
     /// The filter for task name (prefixed).
     ///
     /// If unspecified, no task name filtering is done.
+    #[cfg_attr(feature = "serde", serde(skip_serializing_if = "Option::is_none"))]
     pub name_prefix: Option<String>,
     /// The filter for task state.
     ///
     /// If unspecified, no task state filtering is done.
+    #[cfg_attr(feature = "serde", serde(skip_serializing_if = "Option::is_none"))]
     pub state: Option<State>,
     /// The filter for task tag keys.
     ///
     /// This is zipped with `tag_values`.
     ///
     /// If empty, no task tags filtering is done.
-    #[cfg_attr(feature = "serde", serde(rename = "tag_key", default))]
+    #[cfg_attr(
+        feature = "serde",
+        serde(rename = "tag_key", default, skip_serializing_if = "Vec::is_empty")
+    )]
     pub tag_keys: Vec<String>,
```

Comment:
Should this be an `Option<Vec<String>>` just to be explicit that the field is optional? I don't think practically it's going to have much impact, but perhaps it will to some implementations.